accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,502 @@
+    .NET platform the consumer is now able to upgrade the .NET platform and the
+    OS bindings independently, which makes it hard to explain what combinations
+    are supported. In practice, the OS bindings will want to depend on .NET
+    types which might change over time (think `Task<T>` or `Span<T>)`, so not
+    every combination can work.
+
+We believe it's much easier if we enable code to use multi-targeting (i.e.
+compile the same code for multiple platforms, like we do today.
+
+### Is .NET 5 a superset of .NET Framework 4.x?
+
+No, .NET 5 isn't a *superset* of .NET Framework. However, .NET 5 is
+the *successor* of both .NET Core 3.x as well as .NET Framework 4.x. Starting
+with .NET Core 3, you can build the same kind of workloads that you can with
+.NET Framework. Some of the tech has changed, but that's the way it's [going to
+be forever](https://devblogs.microsoft.com/dotnet/net-core-is-the-future-of-net/).
+The remaining delta will never be closed. Existing apps can stay on .NET
+Framework and will be supported, but we already said we'll no longer add new
+features to it.
+
+Thus, new apps should start on .NET Core. By branding it as .NET 5, this
+recommendation is much more obvious to both, existing customers, as well as new
+customers.
+
+### Why are the OS specific flavors not versioned by the OS?
+
+There are two reasons why this isn't desirable:
+
+1.  **It results a combinatorial explosion**. A TFM in the form of `net5-win7`
+    would (syntactically) make any combination of .NET and OS possible. This
+    raises the question which combinations are supported, which puts us back
+    into having to provide the customer with a decoder ring.
+
+2.  **It can make asset selection ill-defined**. Suppose the project is
+    targeting `net7-win10`. The package offers `net5-win10` and `net6-win7`. Now
+    neither asset would be better.
+
+Developers will want to target different OS versions from a single code
+base/NuGet package, but that doesn't mean they will need to use multi-targeting.
+Multi-targeting is a very heavy hammer. Yes, many people are using it to target
+different versions of .NET (e.g. `net45` vs `net461`). But that's not
+necessarily because `#if` is the better experience, it's because it's simply not
+possible any other way, due to .NET runtime constraints (i.e. assembly
+references and type/member references need to be resolved by the JIT). This
+problem doesn't exist for OS APIs. Developers can generally build a single
+binary for multiple versions of an operating system. The calls to APIs
+introduced in later versions must be guarded, but this is generally understood,
+and we have some limited tooling already with plans to extend it.
+
+However, we do want to allow the developer to express a minimum version they
+require for the OS. It will likely be a property in the project file that is
+being persisted in the resulting NuGet package as well. There is a separate doc
+that we're working on.
```

Comment:
This is just confusing, above you say "because combinatorial explosion" and then state, we'll going to do it anyway. I think we should map `-win10` -> `Target_Platform_Identifer`/`Target_Platform_Version` and skip the profile concept all together.

Reply:
The point I was trying to make is that we don't want to support OS versioning as part of the friendly name.